### PR TITLE
Newsletters: Email Preview Improvements

### PR DESCRIPTION
**Pull request by lezama**
## Proposed changes:
* Improves Preview Email controls

Screenshot 2024-08-06 at 2 25 44 PM

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* In the editor add `showNewsletterMenu` query arg to see the Newsletter plugin icon.
* Click "Preview email"

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/subscriptions/email-preview.js [hunk 1]
```diff
@@ -112,3 +117,155 @@ export default function EmailPreview( { isModalOpen, closeModal } ) {
...
+const DevicePicker = ( { selectedDevice, setSelectedDevice } ) => (
+	<ButtonGroup>
+		{ devices.map( device => (
+			<Button
+				key={ device.name }
+				icon={ device.icon }
+				isSmall
+				isPressed={ selectedDevice === device.name }
+				onClick={ () => setSelectedDevice( device.name ) }
+				style={ {
```
Comment (enejb) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/email-preview.js:137]: Should this be a class? and these values come from the stylesheet?

Comment (lezama) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/email-preview.js:137]: I am trying to use inline on the components that allow for it

File: projects/plugins/jetpack/extensions/blocks/subscriptions/email-preview.js [hunk 2]
```diff
@@ -112,3 +117,151 @@ export default function EmailPreview( { isModalOpen, closeModal } ) {
...
+const devices = [
+	{ name: 'desktop', icon: desktop, label: __( 'Desktop', 'jetpack' ), width: '100%' },
+	{ name: 'tablet', icon: tablet, label: __( 'Tablet', 'jetpack' ), width: '768px' },
+	{ name: 'mobile', icon: mobile, label: __( 'Mobile', 'jetpack' ), width: '360px' },
+];
+
+const DevicePicker = ( { selectedDevice, setSelectedDevice } ) => (
+	<ButtonGroup>
+		{ devices.map( device => (
+			<Button
```
Comment (enejb) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/email-preview.js:130]: I think we should add a label to these buttons for improved accessibilty. See label https://developer.wordpress.org/block-editor/reference-guides/components/button/

**Comment by enejb** [comment]
I don't think we need to address this as part of this PR but it would be good to have a better error state. 

Screenshot 2024-08-06 at 10 51 07 AM

Two things that I see here. 
1. The scroll ( do we need it ?) 
2. The error is not actionable.

**Comment by lezama** [comment]
> The scroll ( do we need it ?)
> The error is not actionable.

I still need to go through the design of that part :)

**Comment by enejb** [comment]
I found a few of issues.
That I think we need to address. 

Double scroll bars. See 
Screenshot 2024-08-06 at 11 03 00 AM

On mobile there are a couple more. 

The icon size and alignment seems off. 
Screenshot 2024-08-06 at 11 04 10 AM

The choice of different resolution icons. 
I think we can remove them. 
See 
Screenshot 2024-08-06 at 11 04 27 AM

There is also a double scroll bar here. Which can be a frustrating experience. Depending on where you scroll you might not be able to.

**Comment by enejb** [comment]
I wonder if we should detect if there are any blocks that would offered a different content and only show users make it so that different things get loaded based on the presence of the content.

**Comment by enejb** [comment]
Could we add something about unsaved changes. Since the preview will not show anything new unless you save the changes before opening the preview.

**Comment by lezama** [comment]
> Could we add something about unsaved changes. Since the preview will not show anything new unless you save the changes before opening the preview.

we discussed this a little bit Yesterday, we need to look into saving a revision or ask the user somehow to save

**Comment by lezama** [comment]
> The icon size and alignment seems off.

solved

**Comment by lezama** [comment]
@enejb tracked your observations here https://github.com/Automattic/jetpack-roadmap/issues/1761, fee free to add!

I think we can merge and follow up in smaller PRs

**Review by enejb — APPROVED** [review]
Thanks for all the fixes!

**Pull request merged**